src/content/docs/developer-guide/git-guide.md

```diff
@@ -0,0 +1,107 @@
+```
+
+``HEAD`` refers to the latest commit of the main branch, and ``x`` represents the number of commits that are on the branch. To find out what this number is you will need to run:
+
+```git
+git log --branches[=<branch>]
+```
+
+The ``--branches`` flag is useful when you’re working on a project that has a lot of commits. It allows you to check only the commits made on your branch.
+
+Now when you run the interactive rebase command, ``git`` will return a file that contains all of your commits. 
+
+In this file, you will see all of the commits, with the word “pick” in front of them. To combine the commits, change all the “pick” words, except for the first one, to “squash”.
+
+After you save and close the file, a new one will open with all of the combined commit messages. All you have to do here is write the new commit message. This will squash your commits and generate a new commit message.
+
+For more in-depth info on the ``rebase`` command, here is a link to [its entry in the reference guide](https://git-scm.com/book/en/v2/Git-Branching-Rebasing). You can also check out this [tutorial](https://www.atlassian.com/git/tutorials/rewriting-history/git-rebase) for some extra help.
+
+## git commit
+
+Another way to avoid many small commits that cause conflicts is through the ``git commit --amend`` command. For example, if you make a small change to a file or a mistake in the commit message, you can use ``--amend`` for a quick fix.
+
+To change the commit message using ``--amend``, all you need to do is add the ``-m`` flag:
+
+```git
+git commit --amend -m "updated commit message”
+```
+
+To add an extra file to your previous commit, you will first need to stage it and then simply run the ``git commit --amend`` command. You can use the ``--no-edit`` flag to amend the commit without modifying the commit message.
+
+```git
+git add newfile.py
+git commit --amend --no-edit
+```
+
+Be careful when using ``--amend`` because it actually creates an entirely new commit, which replaces the one you intend to amend. This can have a confusing effect when used on public commits, since it will modify the branch’s history. Only use ``--amend`` on branches that you are working on by yourself.
+
+You may also run into issues with some pre-commit or commit message hooks. These hooks often run linting, tests, and other types of checks. To skip these tests, you can use the ``--no-verify`` flag, or its shorthand ``-n``.
+
+```git
+git commit --no-verify -m “message”
+```
+
+Learn more about ``commit`` [here](https://git-scm.com/docs/git-commit).
+
+## git push
+
+The ``--no-verify`` flag is also available for the ``git-push`` command, and it serves the same function. If you are running into pre-commit hooks when pushing a branch, try using this flag.
+
+### Solving upstream issues
+
+When you fork a repository, you can avoid a lot of issues by using the following command:
+
+```git
+git remote add upstream https://github.com/name-of-the-original-repo.git
```

Comment:
You may specify the ratatui repo `https://github.com/ratatui-org/ratatui.git` because this guide is specifically about that.